src/java.base/unix/native/libnet/net_util_md.c

```diff
@@ -282,7 +282,7 @@ NET_InetAddressToSockaddr(JNIEnv *env, jobject iaObj, int port,
     } else {
         jint address;
         if (family != java_net_InetAddress_IPv4) {
-            JNU_ThrowByName(env, JNU_JAVANETPKG "SocketException", "Protocol family unavailable");
+            JNU_ThrowByName(env, JNU_JAVANETPKG "SocketException", "Protocol family ipv6, but ipv6 on machine unavailable");
```

Comment:
Hello Matthias, just a drive-by comment. I haven't yet paid close attention to this proposal. If we are going to change this message, then perhaps just change it to `IPv6 protocol family unavailable`. 

Looking at the implementation of `ipv6_available()`, for it to return `false` and thus to throw this exception, either IPv6 isn't supported on that host or the `-Djava.net.preferIPv4Stack=true` system property is set. So I think it's better to leave out the IPv6 not available on machine part.